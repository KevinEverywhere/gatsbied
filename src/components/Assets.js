import React from 'react'
import { withPrefix } from 'gatsby';

class Assets extends React.PureComponent {
  constructor(props){
    super();
    this.state={assets:null}
  }
  componentDidMount(){
    import('aframe')
      .then((aframe) => {
          this.setState({
          assets:(
            <a-assets>
              <video id="haight" webkit-playsinline="" playsInline="" autoPlay="" controls="" src={withPrefix('/videos/satTemptingHaight.mp4')} crossOrigin="anonymous" />
              <video id="biking" webkit-playsinline="" playsInline={true} autoPlay={true} controls="" src={withPrefix('/videos/biking.mp4')} crossOrigin="anonymous" />
              <video id="thankyou" webkit-playsinline="" playsInline="" autoPlay="" controls="" src={withPrefix('/videos/three36.mp4')} crossOrigin="anonymous" />
            </a-assets>
          )})
      })
      .catch((error) => console.error(error));
  }
  render() {
    return this.state.assets;
  }
}

export default Assets
