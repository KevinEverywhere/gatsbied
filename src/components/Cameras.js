import React from 'react'

class Cameras extends React.PureComponent {
  constructor(props){
    super();
    this.state={cameras:null}
  }
  createdAnimations(fromAnimArray, sequential){
    let myAnimations;
    if(!sequential){
      myAnimations = fromAnimArray.map((anim, i) => {
        let str = ` animation__${i}=`
        for(var z in this){
          str += `${z}: ${this[z]};`
        }
        return str.substr(0,str.length-1)
      });
    }
    return myAnimations;
  }
  componentDidMount(){
    import('aframe')
      .then(() => {
        import('aframe-extras')
          .then(() => {
            this.setState({
              cameras:(
                <a-entity id="rig" position="0 10 -3" rotation="0 0 0">
                  <a-camera far="20000" position="0 2 0" id="main-camera">
                    <a-cursor></a-cursor>
                  </a-camera>
                </a-entity>
              )})
          })
      })
      .catch((error) => console.error(error));
    }
  render() {
    return this.state.cameras;
  }
}

export default Cameras
