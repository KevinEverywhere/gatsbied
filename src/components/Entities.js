import React from 'react'
import '../aframe/VideoTexture'

class Entities extends React.PureComponent {
  constructor(props){
    super();
    this.state={entities:null}
  }
  clicked(e){
    console.log(e)
  }
  componentDidMount(){
    import('aframe')
      .then((aframe) => {
          this.setState({
          entities:(
              <a-entity>
                <a-box position="-3 0.5 3" rotation="0 45 0" color="#4CC3D9" />
                <a-sphere position="-10 -1.25 5" radius="1.25" color="#2D5EEF" />
                <a-cylinder position="2 0.75 1" radius="0.5" height="1.5" color="#FFC65D" />
                <a-plane position="0 -1 4" rotation="-90 0 0" width="4" height="4" color="#C87BA4" />
                <a-video src="#skyTexture" video-texture rotation="0 180 0" width="8" height="4" position="-1 0 4">
                <a-animation attribute="rotation" repeat="indefinite" to="0 360 0" dur="3000" easing="linear"></a-animation>
                </a-video>
                <a-entity id="rig" rotation="0 180 0">
                  <a-camera  id="camera">
                    <a-cursor></a-cursor>
                  </a-camera>
                </a-entity>
                <a-sky color="#954" />
              </a-entity>
          )})
      })
      .catch((error) => console.error(error));
  }
  render() {
    return this.state.entities;
  }
}

export default Entities
