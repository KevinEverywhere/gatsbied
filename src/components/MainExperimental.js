import React from 'react'
import Assets from './Assets'
import Cameras from './Cameras'
import Entities from './Entities'

class MainExperimental extends React.PureComponent {
  constructor(props){
    super();
    this.state={
      loadedContent:null,
      aframeLoaded: null,
    }
  }
  componentDidMount(){
    import('aframe')
      .then((aframe) => {
        if(!!!aframe.components['video-texture']){
          aframe.registerComponent("video-texture", {
            init: function() {
              this.el.addEventListener("click", (e) => {
                try{
                  if(document.querySelector(this.el.getAttribute('src')).paused){
                    document.querySelector(this.el.getAttribute('src')).play()
                  }else{
                    document.querySelector(this.el.getAttribute('src')).pause()
                  }
                }catch(oops){}
              })
            }
          })

          aframe.registerComponent("plane-texture", {
            init: function() {
              this.el.addEventListener("click", (e) => {
                let _element = this.el.sceneEl;
                if (document.fullscreenElement || document.mozFullScreenElement || document.webkitFullscreenElement) {
                  if (document.cancelFullScreen) {
                      document.cancelFullScreen();
                  } else {
                      if (document.mozCancelFullScreen) {
                          document.mozCancelFullScreen();
                      } else {
                          if (document.webkitCancelFullScreen) {
                              document.webkitCancelFullScreen();
                          }
                      }
                  }
                } else {
                    if (_element.requestFullscreen) {
                        _element.requestFullscreen();
                    } else {
                        if (_element.mozRequestFullScreen) {
                            _element.mozRequestFullScreen();
                        } else {
                            if (_element.webkitRequestFullscreen) {
                                _element.webkitRequestFullscreen(Element.ALLOW_KEYBOARD_INPUT);
                            }
                        }
                    }
                }
              })
            }
          })
        }
        this.setState({
          aframeLoaded: true,
          loadedContent:(
            <main>
              <a-scene cursor="rayOrigin: mouse" embedded style={{width:'100%',height:'100%'}}>
                <Assets data={this.props.data} />
                <Entities experimental="true" data={this.props.data} />
                <Cameras data={this.props.data} />
                { this.props.children }
              </a-scene>
            </main>
          )})
      })
      .catch((error) => console.error(error));
  }
  render() {
    return this.state.loadedContent;
  }
}

export default MainExperimental
