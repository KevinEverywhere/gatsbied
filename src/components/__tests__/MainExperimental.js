import React from "react"
import renderer from "react-test-renderer"
import MainExperimental from "../MainExperimental"

describe("MainExperimental", () => {
  Promise.all([
    import('../Assets'),
    import('../Cameras'),
    import('../Entities')
  ]).then([ Assets, Cameras, Entities]) => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<MainExperimental />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
