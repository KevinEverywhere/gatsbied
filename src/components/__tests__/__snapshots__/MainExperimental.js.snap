// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainExperimental renders correctly 1`] = `null`;
